src/Number.php

```diff
@@ -345,7 +406,7 @@ private function initFromString($number)
     {
         if (!preg_match("/^(?<sign>[-+])?(?<integerPart>\d+)(?:\.(?<fractionalPart>\d+))?$/", $number, $parts)) {
             throw new \InvalidArgumentException(
-                sprintf('"%s" cannot be interpreted as a number', $number)
+                sprintf('"%s" cannot be interpreted as a number', print_r($number, true))
```

Comment:
Yeah, the problem was that objects and arrays cannot be casted to string automatically, I needed a generic way. I don't think that readability or line breaks are going to be a problem.